prow/jobs/jinja/postsubmits/codegen_release.jinja2

```diff
@@ -25,7 +25,7 @@
             valueFrom:
               secretKeyRef:
                 name: prowjob-github-token
-                key: token
+                key: cert
```

Comment:
I'm not sure if our scripts, which use the Github PAT to authenticate git pull/push, will be able to support using a cert to run their jobs. This needs extra validation

Reply:
Your'e right, [we need to migrate](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-a-github-app) from including the Personal Access Token as part of request URLs ([example](https://github.com/aws-controllers-k8s/test-infra/blob/76e2cacd611c5054baa1636363c7b5e0014efaa4/cd/auto-generate/auto-generate-controllers.sh#L253)) to generating JSON web tokens and include them via an `Authorization:` HTTP header.

From a cursory examination of `test-infra/cd/**` it appears that all of the scripts there are in bash.  Are there any non-bash script-like code that you're aware of that uses the PATs?    @A-Hilaly would appreciate your input here as well.

Reply:
FWIW these are the places I've found so far that appear to use a PAT:
https://github.com/aws-controllers-k8s/community/blob/e001fa05183b32d8d2bcbcfc6c52c9a9864b0eb2/docs/scripts/gen_services.py#L100
https://github.com/aws-controllers-k8s/test-infra/blob/76e2cacd611c5054baa1636363c7b5e0014efaa4/cd/auto-generate/auto-generate-controllers.sh#L253
https://github.com/aws-controllers-k8s/test-infra/blob/2bc4f6fd193e2c6b2d91cc25f93acb976d3e8f37/cd/auto-generate/controller-release-tag.sh#L115
https://github.com/aws-controllers-k8s/test-infra/blob/1447a1d39a79afa23f48e2a5d8b30ecd248ee05b/cd/auto-update/project-static-files.sh#L160
https://github.com/aws-controllers-k8s/test-infra/blob/1bdb5f3c260bb6dcbce92cc6c5bb7c28fc22f30a/cd/olm/olm-bundle-pr.sh#L179
https://github.com/aws-controllers-k8s/test-infra/blob/5d4d92a8c84daf77fb2bad4b03253dd1c91c1f66/prow/jobs/images/build-docs.sh#L43
https://github.com/aws-controllers-k8s/test-infra/blob/81a36ff345e15b0920cc4cb56e0308df6c96a38e/tools/cmd/ack-discover#L56

Reply:
>Are there any non-bash script-like code that you're aware of that uses the PATs?

Not that i'm aware of in `test-infra`, however ack-dev uses PATs in go code to interact with Github.

>Your'e right, [we need to migrate](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-a-github-app) from including the Personal Access Token as part of request URLs ([example](https://github.com/aws-controllers-k8s/test-infra/blob/76e2cacd611c5054baa1636363c7b5e0014efaa4/cd/auto-generate/auto-generate-controllers.sh#L253)) to generating JSON web tokens and include them via an Authorization: HTTP header.

@jljaco for scripts that needs to git pull/push, are you thinking of using tokens with some like: `git push https://<GITHUB_ACCESS_TOKEN>@github.com/<GITHUB_USERNAME>/<REPOSITORY_NAME>.git
`

Reply:
@A-Hilaly I am thinking of [using an installation access token](https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation).  Do you think that would work?

Reply:
Another option is to use the [`content` section of the GitHub API](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents) to perform all of the equivalent actions.    There would be significantly more effort required for that.

Reply:
We (@RedbackThomson, @azpaulp, and I ) discussed this offline, and came to the conclusion that the best way forward here is to leave the PATs in for the sake of the scripts, while still converting Prow to using the GitHub App model.